cylc/flow/scheduler.py

```diff
@@ -1771,6 +1784,8 @@ async def _shutdown(self, reason: Exception) -> None:
             except OSError as exc:
                 LOG.warning(f"failed to remove workflow contact file: {fname}")
                 LOG.exception(exc)
+            else:
+                self.contact_data = None
```

Comment:
What's this do? Could you add a comment.

Reply:
Isn't this straightforward? If the contact file is removed successfully this property is updated to reflect that.

Seeing as we don't normally do anything with the Scheduler after shutdown, the main use for this is testing convenience. It allows checking for `self.contact_data` during tear down of the `run` fixture
https://github.com/cylc/cylc-flow/blob/a917e65948b80590a313b741bb7079437f1d0dcc/tests/integration/utils/flow_tools.py#L146-L149

Reply:
Wouldn't be asking if it was obvious, high chance I'll come along in a few months having forgotten about this and remove it. Worth adding a quick comment so we know this odd line is for the integration tests.